server/storages/mysql.go

```diff
@@ -0,0 +1,254 @@
+	if storeFailedEvents {
+		return tableResults, failedEvents, nil
+	}
+
+	return tableResults, nil, nil
+}
+
+//check table schema
+//and store data into one table
+func (m *MySQL) storeTable(fdata *schema.ProcessedFile, table *adapters.Table) error {
+	_, tableHelper := m.getAdapters()
+	dbSchema, err := tableHelper.EnsureTableWithoutCaching(m.ID(), table)
+	if err != nil {
+		return err
+	}
+
+	start := time.Now()
+	if err := m.adapter.BulkInsert(dbSchema, fdata.GetPayload()); err != nil {
+		return err
+	}
+	logging.Debugf("[%s] Inserted [%d] rows in [%.2f] seconds", m.ID(), len(fdata.GetPayload()), time.Now().Sub(start).Seconds())
+
+	return nil
+}
+
+//SyncStore is used in storing chunk of pulled data to Postgres with processing
+func (m *MySQL) SyncStore(overriddenDataSchema *schema.BatchHeader, objects []map[string]interface{}, timeIntervalValue string, cacheTable bool) error {
+	_, tableHelper := m.getAdapters()
+	flatData, err := m.processor.ProcessPulledEvents(timeIntervalValue, objects)
+	if err != nil {
+		return err
+	}
+
+	deleteConditions := adapters.DeleteByTimeChunkCondition(timeIntervalValue)
+
+	//table schema overridden
+	if overriddenDataSchema != nil && len(overriddenDataSchema.Fields) > 0 {
+		var data []map[string]interface{}
+		//ignore table multiplexing from mapping step
+		for _, fdata := range flatData {
+			data = append(data, fdata.GetPayload()...)
+			//enrich overridden schema with new fields (some system fields or e.g. after lookup step)
+			overriddenDataSchema.Fields.Add(fdata.BatchHeader.Fields)
+		}
+
+		table := tableHelper.MapTableSchema(overriddenDataSchema)
+
+		dbSchema, err := tableHelper.EnsureTable(m.ID(), table, cacheTable)
+		if err != nil {
+			return err
+		}
+		start := time.Now()
+		if err = m.adapter.BulkUpdate(dbSchema, data, deleteConditions); err != nil {
+			return err
+		}
+		logging.Debugf("[%s] Inserted [%d] rows in [%.2f] seconds", m.ID(), len(data), time.Now().Sub(start).Seconds())
+
+		return nil
+	}
+
+	//plain flow
+	for _, fdata := range flatData {
+		table := tableHelper.MapTableSchema(fdata.BatchHeader)
+
+		//overridden table destinationID
+		if overriddenDataSchema != nil && overriddenDataSchema.TableName != "" {
+			table.Name = overriddenDataSchema.TableName
+		}
+
+		dbSchema, err := tableHelper.EnsureTable(m.ID(), table, cacheTable)
+		if err != nil {
+			return err
+		}
+		start := time.Now()
+		if err = m.adapter.BulkUpdate(dbSchema, fdata.GetPayload(), deleteConditions); err != nil {
+			return err
+		}
+		logging.Debugf("[%s] Inserted [%d] rows in [%.2f] seconds", m.ID(), len(fdata.GetPayload()), time.Now().Sub(start).Seconds())
+	}
+
+	return nil
+}
+
+//Update uses SyncStore under the hood
+func (m *MySQL) Update(object map[string]interface{}) error {
+	return m.SyncStore(nil, []map[string]interface{}{object}, "", true)
+}
+
+//GetUsersRecognition returns users recognition configuration
+func (m *MySQL) GetUsersRecognition() *UserRecognitionConfiguration {
```

Comment:
You can omit: `IsCachingDisabled()`, `IsStaging `, GetUniqueIDField `, because they are in Abstract.

Reply:
removed these methods